src/co_tools/co_fastq.py

```diff
@@ -0,0 +1,156 @@
+        )
+        return 0
+
+
+def get_fwd_fastqs(dir: str = "../data"):
+    if some_fastq := (
+        subprocess.check_output(["find", "-L", dir, "-name", "*.fastq.gz"])
+        .decode("utf-8")
+        .strip()
+    ):
+        if "\n" in some_fastq:
+            some_fastq = some_fastq.split("\n")[0]
+        LOGGER.debug(f"some_fastq: {some_fastq}")
+        pattern = get_read_pattern(some_fastq)
+        LOGGER.debug(f"pattern: {pattern}")
+        subprocess_files = (
+        subprocess.check_output(["find", "-L", dir, "-name", f"*{pattern}"])
+        .decode("utf-8")
+        .strip()
+    )
+        LOGGER.debug(f"subprocess_files:\n{subprocess_files}")
+        subprocess_files = subprocess_files.split("\n")
+        subprocess_files.sort()
+        LOGGER.debug(f"sorted subprocess_files:\n{subprocess_files}")
+        files = ""
+        for elem in files:
+            files += f"{elem}\n"
+        files = files.strip()
+        LOGGER.debug(f"returning files:\n{files}")
+        return files
+    else:
+        LOGGER.error(f"There are no fastq.gz files in the {dir} directory")
+        return 0
+
+
+def get_read_direction(filepath: str):
+    filename = filepath.split("/")[-1]
+    LOGGER.debug(f"filename: {filename}")
+    if "_" not in filename:
+        LOGGER.warning(
+            "You might be trying to use a single end reads file as a paired"
+            + f" end reads file. Current input: {filepath}"
+        )
+        return 0
+    return "1" if "1" in filename.split("_")[-1].split(".")[0] else "2"
+
+
+def get_read_pattern(filename: str, direction: str = "1"):
+    if "_" not in filename and "/" in filename:
+        LOGGER.warning(
+            f"{filename} might be a single end reads file. The pattern being returned"
+            + " is the entire filename"
+        )
+        return filename.split("/")[-1]
+    direction_complement = "2" if direction == "1" else "1"
+    pattern = filename.split("_")[-1]
+    LOGGER.debug(f"pattern: {pattern}")
+    return (
+        pattern
+        if direction in pattern
+        else pattern.replace(direction_complement, direction)
+    )
+
+
+def get_prefix(filename: str, split_position: str = "-1"):
+    filename = Path(filename).name
+    # illumina read files use a specific format, and sometimes allow underscores in the prefix
+    # SampleName_S1_L001_R1_001.fastq.gz for lane 1
+    # SampleName_S1_R1_001.fastq.gz for merged lanes.
+
+    if match := re.search(r"(.*?)_S\d+_.*R\d_001.fastq.gz", filename):
+        LOGGER.debug(f"match: {match}\ngroup 1 (prefix): {match.group(1)}")
+        return match.group(1)
+
+    if "_" in filename and int(split_position):
+        prefix_list = filename.split("_")[: int(split_position)]
+        LOGGER.debug(f"prefix_list: {prefix_list}")
+        return "_".join(prefix_list)
+
+    return filename.split(".")[0]
+
+
+def get_rev_file(fwd_file: str, name_only=False):
```

Comment:
This isn't going to work in most cases, fastq files generally don't end with the read direction, especially if they use Illumina demultiplexing. Part of the reason all current capsules allow people to specify the suffix explicitly is because you can't assume that pattern. This would be more useful if it simply took in static forward and reverse suffixes instead of trying to get them from the file name.

Reply:
What exactly is the code suggestion here?

Reply:
Added optional args for specifying the patterns. Retaining the ability to calculate the patterns when user doesn't provide the patterns.